### DISCUSSION TITLE
Intercom: refractory period between detected rings

### DISCUSSION THREAD

**Discussion by andrew-rinato** [dgreif/ring#1369] [Show and tell]
Hi all, I think I found out how to configure the refractory period that prevents subsequent rings from being detected on the intercom. This was a topic of a number of comments on this thread (https://github.com/dgreif/ring/issues/1083#issuecomment-1347870980) and around the web.

I managed to get my system to detect rings faster by dropping the setting `{"video_settings": {"clip_length_max": 60}}` from its default of 60 to 10.

This was helpful for us because we have two doors controlled by the same opening switch. We would arrive at the second before the default refractory period was up, preventing us from opening the door via the app.

Code is here https://github.com/tchellomello/python-ring-doorbell/pull/314

Thanks for all work on this project.

**Comment by tsightler** [comment]
It's pretty strange that a device without video recording somehow uses video settings to control the cool-down period between notifications.  I guess it might make some sense if they are using a shared code path with the doorbell, but still weird.

**Reply by andrew-rinato** [reply]
Yes I think that's what's happening, _that considering their history of devices with recording enabled, the max frequency of detection events is determined by the recording duration (because these events trigger recordings). And although the intercom doesn't have recording, they might be using this parameter to set ring detection frequency_. ~~although it's not meant explicitly to catch redundant rings, but to prevent overlapping recordings. I thought they might be using existing functionality to control recording frequency from the intercom, and since all their other devices have cameras (?), the recording is called video. At least that's how I made sense of it.~~ I didn't exhaustively try all the settings, but I tried about 4 that seemed to have potential. It could be that I stumbled on a solution that isn't the most direct route.

**Reply by tsightler** [reply]
I didn't realize the Ring intercom supported recording the audio at all.  Cameras behave the same way, but with cameras you can set the recording behavior in the Ring app, is it possible to change the recording length in the Ring app for the intercom?   I don't believe it would be the correct behavior to just silently change the recording length for the user.

**Reply by andrew-rinato** [reply]
I misspoke above - you're right that it seems there aren't recordings for the Intercom (as far as I can tell, but I am currently testing to try and get the streaming audio feeds).

To correct what I wrote before - it would make sense that the interval between detected events is limited by the recording duration for devices that start recording on these triggers. I think this could be why the intercom uses this interval to set the max detection frequency even though it doesn't save recordings (in other words, we agree they are probably reusing the existing code path).
